modules/core/src/smithy4s/Service.scala

```diff
@@ -34,33 +34,48 @@ import kinds._
   *   around makes it drastically easier to implement logic generically, without involving
   *   metaprogramming.
   */
-trait Service[Alg[_[_, _, _, _, _]], Op[_, _, _, _, _]] extends FunctorK5[Alg] with Service.Provider[Alg, Op] {
-  implicit val serviceInstance: Service[Alg, Op] = this
-  val service = this
+trait Service[Alg[_[_, _, _, _, _]]] extends FunctorK5[Alg] with Service.Provider[Alg] {
+  type Operation[I, E, O, SI, SO]
+  type Endpoint[I, E, O, SI, SO] = smithy4s.Endpoint[Operation, I, E, O, SI, SO]
+  type Interpreter[F[_, _, _, _, _]] = PolyFunction5[Operation, F]
+  type FunctorInterpreter[F[_]] = PolyFunction5[Operation, kinds.Kind1[F]#toKind5]
+  type BiFunctorInterpreter[F[_, _]] = PolyFunction5[Operation, kinds.Kind2[F]#toKind5]
```

Comment:
the type aliases above are path-dependant (by virtue of depending on the Operation abstract type member) , it means we can ascribe `service.Endpoint`, `service.Interpreter`, etc etc. This allows to clean up a number of verbose ascriptions here and there.